frontend/src/App.res

```diff
@@ -39,26 +44,35 @@ let getTestMetrics = (item: GetBenchmarks.t_benchmarks): BenchmarkTest.testMetri
   }
 }
 
-module BenchmarkBranchResults = {
+let getLatestMasterEntry = (~testName, benchmarks) => {
+  BeltHelpers.arrayFindRev(benchmarks, (item: GetBenchmarks.t_benchmarks) => {
+    item.pull_number == None && item.test_name == testName
+  })
+}
+
+module BenchmarkResults = {
   @react.component
-  let make = (~benchmarks: array<GetBenchmarks.t_benchmarks>, ~branch, ~synchronize) => {
-    let dataForBranch = Belt.Array.keep(benchmarks, item => {
-      let itemBranch =
-        item.branch
-        ->Belt.Option.map(DataHelpers.getBranchName)
-        ->Belt.Option.getWithDefault("master")
-      Js.log(itemBranch)
-      itemBranch == branch
+  let make = (~benchmarks: array<GetBenchmarks.t_benchmarks>, ~pullNumber=?, ~synchronize) => {
+    let data = Belt.Array.keep(benchmarks, (item: GetBenchmarks.t_benchmarks) => {
+      // pullNumber is assumed to be None only for master
+      item.pull_number == pullNumber
```

Comment:
Is this a temporary solution because we can't assume that the main branch's name is always equal to "master"?

Reply:
This is ok while we *only* support running benchmarks for pull requests and master.

If we decide to run benchmarks for branches that might not have a PR, this will need to change.

So the assumption here is that if there isn't a PR number it has to be "master" (not the literal `master`, but the default branch for a repo). I'm not sure though what would happen for repos that don't use `master` at the moment.

Happy to clarify this in the code comment if this is not clear.

Reply:
Ok, thanks for the clarifications. I was indeed wondering what would happen for branches that don't have PR. But since benchmarks are not run against them for now that's fine I guess :)